File: backend/src/main/kotlin/com/example/webflux/domain/model/Goods.kt
package com.example.webflux.domain.model

data class Goods(
    val id: String,
    val name: String,
    val description: String,
    val price: Double,
    val media: List<Media>,
    val categoryId: String,
    val difficulty: String,
    val duration: Int? = null,
    val videoUrl: String? = null,
    val previewUrl: String? = null,
    val detailedDescription: String? = null,
    val careInstructions: String? = null
)

File: frontend/src/types/goods.ts
import { Category } from '../services/categoryService';

export type MediaType = 'image' | 'video';

export interface Media {
    type: MediaType;
    url: string;
    order: number;
}

export interface Goods {
    id: string;
    name: string;
    description: string;
    price: number;
    media: Media[];
    categoryId: string;
    category?: Category;
    difficulty: string;
    duration?: number | null;
    videoUrl?: string | null;
    detailedDescription?: string | null;
    careInstructions?: string | null;
}
